### DISCUSSION TITLE
[unreal]ts可以继承自己写的ts类吗？

### DISCUSSION THREAD

**Discussion by stevenBZ** [Tencent/puerts#1168] [Q&A]
比如我已经用ts实现了一个actor类，然后我希望继承自己写的ts类做一些个性化处理，这个是支持的吗？ 我自己测试了一下，把继承的ts类拖入关卡运行编辑器就会崩溃。

代码如下

`import * as UE from 'ue';
import {$ref, $unref} from 'puerts';
import { edit_on_instance } from 'ue';
import TS_PowerUp_Collection from './TS_PowerUp_Collection';
import './ObjectExt';

class TS_PowerUp_Collection_Copy extends TS_PowerUp_Collection
{

  Constructor()
  {
    super.Constructor();
  }

  ReceiveBeginPlay(): void
  {
      super.ReceiveBeginPlay();
  }

  ReceiveTick(DeltaTime: number): void {
    super.ReceiveTick(DeltaTime);
  }
}

export default TS_PowerUp_Collection_Copy;`

**Comment by chexiongsheng** [comment]
可以，之前的fps例子就有

**Reply by stevenBZ** [reply]
感谢🙏
